pkg/ttl/ttlworker/session_integration_test.go

```diff
@@ -97,19 +118,12 @@ func (s *sessionWithFault) ExecuteInternal(ctx context.Context, sql string, args
 }
 
 func (s *sessionWithFault) shouldFault(sql string) bool {
-	if s.fault.Load() == nil {
-		return false
-	}
-
-	// as a fault implementation may have side-effect, we should always call it before checking the SQL.
-	shouldFault := (*s.fault.Load()).shouldFault()
```

Comment:
When `s.fault.Load() != nil`, the `*s.fault` can still be a nil interface.

Reply:
Can we modify setFault?

```
func (f *faultSessionPool) setFault(ft fault) {
	if ft == nil {
		f.fault.Store(nil)
	}
	f.fault.Store(&ft)
}
```